src/zabapgit_repo_impl.prog.abap

```diff
@@ -381,12 +385,17 @@ CLASS lcl_repo IMPLEMENTATION.
           ls_item  TYPE ty_item,
           lt_files TYPE ty_files_tt.
 
+    DATA: lt_cache TYPE SORTED TABLE OF ty_file_item
+          WITH NON-UNIQUE KEY item.
+
     FIELD-SYMBOLS: <ls_file>   LIKE LINE OF lt_files,
                    <ls_return> LIKE LINE OF rt_files,
+                   <ls_cache>  LIKE LINE OF lt_cache,
                    <ls_tadir>  LIKE LINE OF lt_tadir.
 
 
-    IF lines( mt_local ) > 0.
+    " Serialization happened before and no refresh request
+    IF mv_last_serialization IS NOT INITIAL AND mv_do_local_refresh = abap_false.
```

Comment:
can the number of variables be reduced? I guess mv_last_serialization is set if lines( mt_local ) > 0, or there are no objects in the package, which means its okay to run serialization for nothing.

mv_do_local_refresh: just have it clear mt_local, then it can automatically refresh?